### ISSUE TITLE
Limiting memory consumption (Feature request)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [acidburn0zzz/dokan#174]
```
Since Dokan in nature is a Virtual File System (VFS) and thus resides in 
memory, it would be desirable to force a limit on the amount of memory which is 
consumed.

Currently, reading and writing to/from a file is potentially very memory 
consuming due to buffers being allocated. This can obviously be resolved 
programmatically by chopping up large reads and writes in the applications 
using your VFS. Lazy programmers (like me) don't like to write this kind of 
trivial code.

Let me illustrate the issue with an example;
Let's say I have 500 MB of data I want to write to a file in my Dokan-powered 
VFS. I do this by calling the .NET method "File.WriteAllBytes(filename, data)". 
The VFS will then allocate a 500*1024*1024 array of bytes and then perform the 
write-command. 

Now imaging this scenario taking place on a netbook with only 1GB RAM. *ouch*

But ... I gets even worse with the .NET binding, since this relies on the .NET 
Garbage Collector, which is ... erhm ... random at best. OutOfMemory exceptions 
occur frequently (handling large files and/or stress testing) unless one 
forcefully calls the garbage collector regularly.

So, to sum up: some options for maximum read and write buffer sizes would be an 
awesome addition. 

P.S. FUSE has these options, but they are upper-bound by 32 pages of 4096 bytes 
- which I think is too low.

```

Original issue reported on code.google.com by `tokeboy.riis` on 8 Aug 2010 at 7:43
